test/unit/cuda/test-atomic.cpp

```diff
@@ -0,0 +1,257 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
```

Comment:
Why separate CPU and cuda tests? Especially for auto_atomic it seems worthwhile to have the same typed test harness, or similar, exercise both sides.